assignment1/test_string_permutation.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/python3.5
+import unittest
+import string_permutation as pmt
+
+class TestStringPermutations(unittest.TestCase):
+    def setUp(self):
+        self.apple1 = "Apple"
```

Comment:
I get the intention of creating symbolic names for the test strings to avoid typos messing up test cases. It's a valid thing to do.

I would not bother and would use literal strings to make the test cases below easier to follow. But it really does not matter much here and is up to the taste of the one writing the code (you).

I'm mentioning this because test cases can be slightly different than the rest of the code. For example, people occasionally accept a bit more code duplication in test cases if it makes it clearer what the test case is doing. One way to think of it is - since one generally has no test cases to test the test cases themselves, it is imperative for you to be able to look a test case and immediately see that they are doing the right thing.